main.tf

```
@@ -29,4 +42,25 @@ module "zone" {
 
   name = var.name
   tags = var.tags
+  vpc_id = {
+    vpc_id = var.vpc_id.vpc_id
+  }
+
+}
+
+module "vpc_association" {
+  source = ".//modules/vpc_association"
+  count  = var.create_vpc_association ? 1 : 0
```

Changing the vpc input shape, I kinda think we can remove the need for `create_vpc_association` entirely. Just test var.vpc for null...

```suggestion
  count  = var.vpc != null ? 1 : 0
```